### PR TITLE
fix(app-cmds): before and after invoke hooks not working for function inside cogs

### PR DESCRIPTION
**Pull request by okay1204**
## Summary

Fixes an issue where the `before_invoke` and `after_invoke` decorators were not able to be used on functions inside Cogs because of the `self` parameter not being addressed. This fix simply checks how many arguments the passed in hook has and decides if the parent cog should be passed in for the self argument or not.

## Checklist

- [X] If code changes were made then they have been tested.
  - [ ] I have updated the documentation to reflect the changes. 
  - [ ] I have run `task pyright` and fixed the relevant issues.
- [X] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: nextcord/application_command.py [hunk 1]
```diff
@@ -904,13 +904,22 @@ async def invoke_callback_with_hooks(
...
+                    await self._invoke_hook_param_check(after_invoke, interaction)
+
+    async def _invoke_hook_param_check(
+        self, hook: ApplicationHook, interaction: Interaction
+    ) -> None:
+        # invoke with self parameter if it exists
+        if len(signature(hook).parameters) == 2:
+            await hook(self.parent_cog, interaction)  # type: ignore
+        else:
+            await hook(interaction)  # type: ignore
```
Comment (teaishealthy) [inline nextcord/application_command.py:922]:
```suggestion
        await hook(interaction)  # type: ignore
```

**Review by teaishealthy — DISMISSED** [review]

**Comment by okay1204** [comment]
@teaishealthy Reverted the previous change because it broke the fix. It needs the else statement or else the hook is called again with just the interaction parameter.

**Comment by teaishealthy** [comment]
> Reverted the previous change because it broke the fix. It needs the else statement or else the hook is called again with just the interaction parameter.

Yeah I completely forgot the return in the if statement

**Review by Skelmis — APPROVED** [review]
Untested but overall looks fine to me

**Comment by alentoghostflame** [comment]
@okay1204 If you fix the merge conflicts, I'll look into merging this in.

**Review by ooliver1 — APPROVED** [review]

**Pull request merged**